lib/iris/analysis/_regrid.py

```diff
@@ -543,3 +544,336 @@ def __call__(self, src):
                                    sample_grid_x, sample_grid_y,
                                    regrid_callback)
         return result
+
+
+class UnstructuredProjectedRegridder(object):
```

Comment:
it seems to me that there is code sharing taking place between this class and the rectilinear class

is this the case

how much effort is it to refactor to deliver a base class and two specialisations?